libc/include/llvm-libc-macros/limits-macros.h

```diff
@@ -148,7 +148,7 @@
 #endif // INT_MAX
 
 #ifndef UINT_MAX
-#define UINT_MAX (~0U)
+#define UINT_MAX (INT_MAX * 2U + 1U)
```

Comment:
shall we explicitly cast `((unsigned) INT_MAX)`?

Reply:
Should be handled by the `2U` part, otherwise it would fail in the myriad templates we use in `limits.h` I believe.